modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala

```diff
@@ -548,6 +551,280 @@ object ElasticQuery {
   final def ids(value: String, values: String*): IdsQuery[Any] =
     Ids(values = Chunk.fromIterable(value +: values))
 
+  /**
+   * Constructs an instance of [[zio.elasticsearch.query.IntervalMatch]] interval query.
+   *
+   * This query matches analyzed text within specified intervals based on the provided query string.
+   *
+   * @param query
+   *   the text to match in the intervals.
+   * @return
+   *   an instance of [[zio.elasticsearch.query.IntervalMatch]] that represents the `match` interval query.
+   */
+
+  final def intervalMatch(query: String): IntervalMatch[String] =
```

Comment:
We should have specific trait here as return type for methods here. (you can see other methods)